[docs] Add info on building app variants locally with `expo run` (#46543)
expo/expo: docs/pages/build-reference/variants.mdx

> **Note**: If you use EAS Update to publish JavaScript updates of your app, you should be cautious to set the correct environment variables for the app variant that you are publishing for when you run the `eas update` command. See the EAS Build [Environment variables and secrets](/build/updates) for more information.

> ### Build and multiple app variants locally

Author: **[suggestion]** This heading is missing a verb and does not parse as written (`expo-docs-style-guide.md#voice-and-tone`). The section covers building and running variants with `expo run`, so name both actions.

```suggestion
### Build and run multiple app variants locally
```

If you [build your app locally](/guides/local-app-overview/) with [`expo run:android|ios`](/workflow/continuous-native-generation/#usage), set the `APP_VARIANT` environment variable when you run the `expo run` command. For example, to compile the `development` variant for iOS as a debug build, run:

<Terminal
  cmd={{
    npm: ['$ APP_VARIANT=development npx expo run:ios'],
    yarn: ['$ APP_VARIANT=development yarn expo run:ios'],
    pnpm: ['$ APP_VARIANT=development pnpm expo run:ios'],
    bun: ['$ APP_VARIANT=development bun expo run:ios'],
  }}
/>

> `APP_VARIANT` changes only the app's name and package name on Android and bundle identifier on iOS. It does not affect how the binary is compiled. When you switch variants, the existing **android** and **ios** directories still reflect the previous variant. If a native directory is already present, `expo run` will skip regenerating it and compile it as-is.

Author: **[suggestion]** Use present tense instead of "will" (`expo-docs-style-guide.md#voice-and-tone`). The guide convention is "Expo Router resolves the path", not "will resolve".

```suggestion
`APP_VARIANT` changes only the app's name and package name on Android and bundle identifier on iOS. It does not affect how the binary is compiled. When you switch variants, the existing **android** and **ios** directories still reflect the previous variant. If a native directory is already present, `expo run` skips regenerating it and compiles it as-is.
```

> To switch variants, you need to regenerate the native directories again with `prebuild --clean` before compiling the app. In the example below, the `development` variant (existing debug build) is switched to `test`, which is a another debug build. To regenerate the **Android** and **ios** directories and then compile the `test` variant, set `APP_VARIANT` on both commands so that the clean prebuild and the compilation use the same variant:

Author: **[nit]** Two small fixes on this line (`expo-docs-style-guide.md#formatting-and-structure`): "a another" is a typo for "another", and the directory name should be lowercase **android** to match **ios** next to it and the **android** usages on lines 170 and 199.

```suggestion
To switch variants, you need to regenerate the native directories again with `prebuild --clean` before compiling the app. In the example below, the `development` variant (existing debug build) is switched to `test`, which is another debug build. To regenerate the **android** and **ios** directories and then compile the `test` variant, set `APP_VARIANT` on both commands so that the clean prebuild and the compilation use the same variant:
```
